packages/workflow/src/records/handler/unassign.ts

```diff
@@ -45,6 +72,64 @@ export async function unassignRecordHandler(
     ]
   )
 
+  const allTasks = sortTasksDescending(
+    record.entry.map(({ resource }) => resource).filter(isTaskOrTaskHistory)
+  )
+
+  // declaration is unassigned when status changes
+  const latestChangedStatusIndex = allTasks.findIndex(
+    (entry, index, array) =>
+      index < array.length - 1 &&
+      getTaskBusinessStatus(entry) !== getTaskBusinessStatus(array[index + 1])
+  )
+
+  const unassignIndex = findTaskIndexByExtension(
+    allTasks,
+    'http://opencrvs.org/specs/extension/regUnassigned'
+  )
+
+  const lastUnassignedTaskIndex = getLastUnassignedTaskIndex(
+    latestChangedStatusIndex,
+    unassignIndex
+  )
+
+  const lastAssignedTaskIndex = findTaskIndexByExtension(
+    allTasks,
+    'http://opencrvs.org/specs/extension/regAssigned'
+  )
+
+  if (lastAssignedTaskIndex === -1) throw new Error('Not assigned yet')
+
+  // check for unassigned task and status change
+  if (
+    lastUnassignedTaskIndex !== -1 &&
+    lastAssignedTaskIndex > lastUnassignedTaskIndex
+  )
+    throw new Error('Declaration is found unassigned already')
+
+  const task = allTasks[lastAssignedTaskIndex]
+
+  const regLastUser = findExtension(
+    'http://opencrvs.org/specs/extension/regLastUser',
+    task.extension
+  )
+
+  if (!regLastUser) throw new Error('No user is found assigned to this record')
```

Comment:
@Nil20 we can now use the [findAssignment](https://github.com/opencrvs/opencrvs-core/blob/cac7459d7f49eee7aff493aa3a502b95f044b54d/packages/commons/src/assignment.ts#L29) function that Pyry created